### PR TITLE
Feature: App-917

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- Add Ens Validation on DAO creation process

Task: [APP-917](https://aragonassociation.atlassian.net/browse/APP-917)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

**Review by Fabricevladimir — APPROVED** [review]
Looks good to me

**Comment by WissenIstNacht** [comment]
Thanks for the PR.

For some reason I can not make the validation error pop up when creating a DAO. Could you give me a DAO name that would result in a clash and therefore shows the error?

**Comment by sepehr2github** [comment]
> Thanks for the PR.
> 
> For some reason I can not make the validation error pop up when creating a DAO. Could you give me a DAO name that would result in a clash and therefore shows the error?

Hey, There is no way to test this logic because the backend hasn't implemented the ens logic and It's entirely dummy data, I've just put the logic, couldn't tested it even by myself

File: packages/web-app/src/utils/validators.ts [hunk 1]
```diff
@@ -165,3 +167,24 @@ export function actionsAreValid(
...
+    }, 50);
+
+    clearTimeout(timeoutBuffer.current);
+
+    if (ensAddress) return i18n.t('errors.ensDuplication');
+  } catch (err) {
+    return i18n.t('errors.ensNetworkIssue');
+  }
+  return true;
+}
```
Comment (WissenIstNacht) [inline packages/web-app/src/utils/validators.ts]: Yes, it is probably part of the problem as well that the SDK doesn't have that logic implemented. However, I think this code has issues as well.

See, this function is **always** going to return `true`. So even if the SDK's `resolveName` ever did return something useful, your validator function would never make use of it.

It's a bit much to explain in a comment. Let me know if you want to talk about it.

**Review by WissenIstNacht — CHANGES_REQUESTED** [review]

File: packages/web-app/src/utils/validators.ts [hunk 2]
```diff
@@ -165,3 +167,15 @@ export function actionsAreValid(
...
   return isValid;
 }
+
+export async function isDaoNameValid(value: string, provider: InfuraProvider) {
+  if (isOnlyWhitespace(value)) return i18n.t('errors.required.name');
+  let ensAddress = null;
+  try {
+    ensAddress = await provider?.resolveName(value.replaceAll(' ', '_'));
+
+    if (ensAddress) return i18n.t('errors.ensDuplication');
```
Comment (WissenIstNacht) [inline packages/web-app/src/utils/validators.ts:177]: Now you're never returning `true` :D

As far as I remember the true is necessary for when there is no error (or rather, when the input is actually valid). So I think adding `else return true` under this line would make it good to go.

**Review by WissenIstNacht — APPROVED** [review]
Thanks a lot for the changes.

One small modification left, but I'm approving this already :)

File: packages/web-app/src/utils/validators.ts (hunk 2 above)
Comment (sepehr2github) [reply packages/web-app/src/utils/validators.ts:177]: I was about to add it ;D, thanks will fix it

**Pull request merged**